### PR TITLE
Only "module account" transfer events cause vpurse balance updates

### LINKED ISSUE #5896: notifying smart wallet of every vbank balance update is too expensive
## What is the Problem Being Solved?

@michaelfig and I were brainstorming about vat-bank improvements.

As mentioned in #5885, our current vat-bank implementation results in some pretty expensive work being done each time any cosmos-managed balance changes (which includes the RUN and IST balances of every account). The cosmos code sends a VBANK_UPDATE message over the bridge with all the updates, then vat-bank updates an in-RAM copy of those balances. This enables vat-bank to provide a full `Purse` API on the virtual purses it exports to other users.

To avoid that work, we'd really prefer to only use those virtual purses for their `.deposit` and `.withdraw` methods, and leave balance checking (both one-shot and ongoing notification) to more efficient pathways. The cosmos Bank module already publishes all of those balances to the IAVL tree, and there are well-established RPC tools to query them (a balance query being practically the first thing that any blockchain implements).

There are three places that are interested in a balance:

* external users (e.g. a user wallet UI): they should be using the RPC tools to fetch these balances
* the on-chain smart wallet: ideally this should not need to query the cosmos-managed balances at all: it knows of their existence, and it tells the user that they exist, but it should not need to be updated with the actual balance number, except maybe when doing a deposit/withdraw
* an ag-solo (perhaps a wallet within an ag-solo), which might be holding a Presence backed by one of those virtual purses.. we're not sure how to address this yet

The on-chain smart wallet currently subscribes to balance updates for all the user's purses, both the real ones and the virtual purses that reflect cosmos-side Bank module balances. It publishes a single aggregate record to the IAVL tree (containing a list of purses and their current balances) every time any one of those balances change.

We're thinking that that record should only include the "real" purses, not the vat-bank -managed virtual purses. When the wallet UI wants to render a list of all the user's purses, it should make two queries: one to the on-chain wallet's aggregate record (for the "real" purses), and then a second to the Bank module's normal balance-publishing path (for the denominations that would be represented by virtual purses). Then, there should be a shared event, a "doorbell" of some sort. The wallet UI/frontend watches for that event, and when it happens, it re-performs both queries. The on-chain wallet code triggers this event each time it changes the aggregate record (because one of the "real" purses has changed value). The cosmos Bank module (or vbank module?) also triggers this event when one of the vpurse balances has changed.

That way, we're not doing any on-chain work in reaction to a cosmos Bank balance changing: we just ring the doorbell so the frontend can know it should re-query over RPC.

We're thinking that the virtual Purses exported by vat-bank be stripped down: we remove both `getCurrentBalance` (or have it return `undefined` or something), and we remove the `getNotifier` too. The on-chain wallet shouldn't be querying these anymore, and off-chain code should know that the data is available through the normal cosmos Bank RPC queries.

@michaelfig I'm sure I've missed a lot here.. could you help fill in the gaps?

cc @samsiegart @dckc

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #5896

## Description

Reduce SwingSet load by limiting `x/vbank` only to trigger vat-bank balance notifier updates upon explicit virtual purse operations, or on events caused by Cosmos "module account" balance changes.

This works because the smart wallet UI reads Cosmos (vbank) balances directly from the RPC server instead of from the virtual purse notifier.  Uses within Agoric SDK of Cosmos module accounts represented as virtual purses (like the provision pool) work the same as they did before.

Pinging @dckc for the correctness of these assumptions, and @JimLarson for Go changes.

### Scaling Considerations

Improves scalability by reducing SwingSet traffic caused by user-to-user Cosmos-level transfers.

### CODE REVIEW

**Review by dckc — COMMENTED** [review]
Approach seems good

File: golang/cosmos/x/vbank/vbank_test.go [hunk 1]
```diff
@@ -730,16 +746,32 @@ func Test_EndBlock_Rewards(t *testing.T) {
...
+type mockAccount struct{
+	accs map[string]authtypes.AccountI
+}
 
 func (ma mockAccount) GetModuleAddress(name string) sdk.AccAddress {
 	return sdk.AccAddress(name)
 }
 
+func (ma mockAccount) GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI {
+	fmt.Printf("GetAccount %s\n", addr.String())
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/vbank_test.go]: Comment out before merge.

File: golang/cosmos/x/vbank/module.go [hunk 2]
```diff
@@ -155,14 +155,24 @@ NextEvent:
 		}
 	}
 
+	// Prune the addressToUpdate map to only include module accounts.
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/module.go]: Add a note:

```
// We prune only after recording and consolidating all account updates to minimize the number of account keeper queries.
```

File: golang/cosmos/x/vbank/vbank_test.go [hunk 3]
```diff
@@ -730,16 +746,32 @@ func Test_EndBlock_Rewards(t *testing.T) {
 	}
 }
 
-type mockAccount struct{}
+type mockAccount struct{
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/vbank_test.go]: Please name "mockAccountKeeper" or "mockAuthKeeper". (Yes, I know the misleading name predates this PR.)

File: golang/cosmos/x/vbank/vbank_test.go [hunk 4]
```diff
@@ -730,16 +746,32 @@ func Test_EndBlock_Rewards(t *testing.T) {
...
+	return ma.accs[addr.String()]
+}
+
 func Test_Module_Account(t *testing.T) {
-	account := &mockAccount{}
-	keeper, ctx := makeTestKit(account, nil)
-	ch := NewPortHandler(AppModule{}, keeper)
+	acct := &mockAccount{
+		accs: map[string]authtypes.AccountI{
+			"cosmos1we3xzmnt9aex2um9wfmx2em0pd0": &authtypes.ModuleAccount{
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/vbank_test.go]: Consider making the address a named constant for easier visual comparison within the test.

**Review by JimLarson — APPROVED** [review]
Approved with minor fixes.

**Pull request merged**